MailerQ/ApiClient/ApiClient.cs

```diff
@@ -0,0 +1,200 @@
+                _logger.LogError(exception,
+                    "Endpoint: {Endpoint}. Method: {Method}.",
+                    EndpointPause,
+                    MethodDelete);
+                throw exception;
+            }
+        }
+
+        private HttpStatusCode MailerQApiClient(string endpoint, string method, string content)
+        {
+            _apiUri.Path = Path.Combine(_apiUri.Path, endpoint);
+            var request = WebRequest.CreateHttp(_apiUri.Uri);
+
+            request.Method = method;
+            request.Headers.Add("Content-Type", "application/json");
+            request.Headers.Add("Authorization", "Bearer " + _apiClientSettings.Token);
+
+            using (var requestStream = request.GetRequestStream())
+            {
+                using var streamWriter = new StreamWriter(requestStream);
+                streamWriter.Write(content);
+            }
+            var response = (HttpWebResponse)request.GetResponse();
+            return response.StatusCode;
+        }
+
+        public List<Error> Get(Error error)
+        {
+            try
+            {
+                var result = MailerQApiClientGet(EndpointError, MethodGet);
+                return JsonConvert.DeserializeObject<List<Error>>(result);
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception,
+                    "Endpoint: {Endpoint}. Method: {Method}. Exception message: {Message}",
+                    EndpointError,
+                    MethodGet,
+                    exception.Message);
+                throw exception;
+            }
+        }
+
+        public List<Pause> Get(Pause pause)
+        {
+            try
+            {
+                var result = MailerQApiClientGet(EndpointPause, MethodGet);
+                return JsonConvert.DeserializeObject<List<Pause>>(result);
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception,
+                    "Endpoint: {Endpoint}. Method: {Method}.",
+                    EndpointPause,
+                    MethodGet);
+                throw exception;
+            }
+        }
+
+        private string MailerQApiClientGet(string endpoint, string method)
+        {
+            _apiUri.Path = Path.Combine(_apiUri.Path, endpoint);
+            var request = (HttpWebRequest)WebRequest.Create(_apiUri.Uri);
+
+            request.Method = method;
+            request.Headers.Add("Content-Type", "application/json");
+            request.Headers.Add("Authorization", "Bearer " + _apiClientSettings.Token);
+
+            var content = string.Empty;
+
+            using (var response = (HttpWebResponse)request.GetResponse())
+            {
+                using var stream = response.GetResponseStream();
+                using var streamReader = new StreamReader(stream);
+                content = streamReader.ReadToEnd();
+            }
+            return content;
```

Comment:
Here you can intercept the code to allow testing without real requests.

I am not sure if _WebRequest_ is the prefered way to make HTTP request in modern .NET Core apps. If I am not wrong the preferred way is using HttpClient with [IHttpClientFactory](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.ihttpclientfactory?view=dotnet-plat-ext-3.1), it should help you to mock and verify the requests.

I am more familiar with another alternative: https://flurl.dev/

Reply:
We are already using Flurl in Relay, altought the new version offers a better API than the one that we are using in Relay.

Reply:
As @andresmoschini said this should be changed to use *HttpClient*. [*WebRequest* is not recommended](https://docs.microsoft.com/en-us/dotnet/api/system.net.webrequest?view=netstandard-2.0#remarks)